### DISCUSSION TITLE
Uso de preloader em websites (sobre performance e UX)

### DISCUSSION THREAD

**Discussion by leoamrl** [frontendbr/forum#985] [Perguntas e Respostas]
Fala galera, tranquilos?

O que acham sobre o uso de preloader em websites institucionais?
Alguém já fez um teste A/B para medir a Taxa de Rejeição em um site com/sem preloader.
Utilizo em alguns projetos por conta de ter que carregar o CSS no fim do arquivo, defer de imagens, etc.
Caso não seja recomendado, o que seria a melhor solução (ou conjunto de soluções) para um site?

Valeu!

**Comment by linconkusunoki** [comment]
O preloader seria aquele loading até o site carregar tudo?
Se for eu não curto muito, mas é algo pessoal..
Este artigo tem bastante coisa sobre como melhorar a performance sem fazer o preloader
https://css-tricks.com/case-study-boosting-front-end-performance/

Acho que não aborda o server push do protocolo http2, nunca testei esse cara, tem gente que diz que melhora muito mas ele pode ser um pouco complicado.
https://jakearchibald.com/2017/h2-push-tougher-than-i-thought/

**Comment by ninetails** [comment]
Seria isso? [Preload, Prefetch And Priorities in Chrome by Addy Osmani](https://medium.com/reloading/preload-prefetch-and-priorities-in-chrome-776165961bbf)

Nunca utilizei mas acho muito válido o _Critical Path_, ainda mais para SPAs...

Não tenho métricas mas no próprio artigo já tem alguns. Creio que colocar não faz mal, pelo menos para um certo público (Chrome, não sei se no Firefox) pode haver melhorias na performance no primeiro render.

Mas pra sites institucionais, realmente precisa? Claro que pode ser um "a mais", mas pessoalmente acho que a prioridade disto deveria ir depois de SEO, por exemplo. Se o usuário entra é pq ele quer saber do conteúdo, diferente de um e-commerce que a demora no primeiro carregamento/interação pode haver perda de cliente.

**Comment by luizbills** [comment]
Para mim preloader só serve para dizer que seu site é lento. A não ser que o site seja bem "doidão", cheio das animações e realmente precisa de tudo pronto antes, não vejo motivo para usar.

Melhor opção e alternativa, é fazer da maneira correta mesmo: colocar o css no `head`, carregar js com `async`, etc (procura no google sobre _frontend optimizations_ que você vai encontrar muita coisa).

**Comment by ninetails** [comment]
Ah se for preloader, estilo uma animaçãozinha enquanto os assets carregam, sou totalmente contra. Ainda mais se for site institucional.

Tenha como foco principal em carregar o conteúdo primeiro, mesmo que estático, antes mesmo do resto do site carregar.
